src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamServiceAccountProvider.scala

```diff
@@ -17,7 +20,11 @@ abstract class SamServiceAccountProvider(config: Config) extends ServiceAccountP
   implicit val materializer = ActorMaterializer()
   implicit val executionContext = system.dispatcher
 
-  protected lazy val samConfig = config.as[SamConfig]("sam")
-  protected lazy val samDAO = new HttpSamDAO(samConfig.server)
+  protected lazy val samServer = config.as[String]("samServer")
+  protected lazy val cacheExpiryTime = toScalaDuration(Option(config.getDuration("samCacheExpiryTime")).getOrElse(Duration.ofMinutes(60)))
+  protected lazy val cacheMaxSize = Option(config.as[Int]("samCacheMaxSize")).getOrElse(1000)
+  protected lazy val (leoEmail, leoPemFile) = getLeoServiceAccountAndKey
+  protected lazy val samDAO = new HttpSamDAO(samServer)
```

Comment:
We'll need to refactor this to get rid of `HttpSamDAO` and just use `SwaggerSamClient`. Probably for a separate PR though.

Reply:
I'll create a housekeeping ticket for this.

Reply:
https://github.com/DataBiosphere/leonardo/issues/173